example/lambda-at-edge.tf

```diff
@@ -0,0 +1,125 @@
+
+/**
+ * Create the Lambda function. Each new apply will publish a new version.
+ */
+# resource "aws_lambda_function" "lambda" {
+#   function_name = var.name
+#   description   = var.description
+
+#   # Find the file from S3
+#   s3_bucket         = var.s3_artifact_bucket
+#   s3_key            = aws_s3_bucket_object.artifact.id
+#   s3_object_version = aws_s3_bucket_object.artifact.version_id
+#   source_code_hash  = filebase64sha256(data.archive_file.zip_file_for_lambda.output_path)
+
+#   publish = true
+#   handler = var.handler
+#   runtime = var.runtime
+#   role    = aws_iam_role.lambda_at_edge.arn
+#   tags    = module.tags.tags
+
+#   lifecycle {
+#     ignore_changes = [
+#       last_modified,
+#     ]
+#   }
+# }
+
+data "archive_file" "this" {
+  type        = "zip"
+  source_dir  = "${path.module}/src/edge-function"
+  output_path = "${path.module}/src/edge-function.zip"
+}
+
+resource "aws_lambda_function" "this" {
```

Comment:
We shouldn't be hand writing Lambda modules, and I think the Lambda info should be passed into this module as optional settings if it is needed. This module should not have a nested Lambda function inside of it.